pyomo/contrib/doe/fim_doe.py

```diff
@@ -0,0 +1,2774 @@
+                # get right line number in kaug results
+                if self.discretize_model is not None:
+                    # for DAE model, some variables are fixed
+                    try:
+                        kaug_no = col.index(measurement_name)
+                        measurement_index.append(kaug_no)
+                        # get right line of dsdp
+                        dsdp_extract.append(dsdp_array[kaug_no])
+                    except:
+                        if self.verbose:
+                            print('The variable is fixed:', measurement_name)
+                        # for fixed variables, the sensitivity are a zero vector
+                        dsdp_extract.append(zero_sens)
+                else:
+                    kaug_no = col.index(measurement_name)
+                    measurement_index.append(kaug_no)
+                    # get right line of dsdp
+                    dsdp_extract.append(dsdp_array[kaug_no])
+
+            # Extract and calculate sensitivity if scaled by constants or parameters.
+            # Convert sensitivity to a dictionary
+            jac = {}
+            for par in self.param_name:
+                jac[par] = []
+
+            for d in range(len(dsdp_extract)):
+                for p, par in enumerate(self.param_name):
+                    # if scaled by parameter value or constant value
+                    if self.scale_nominal_param_value:
+                        jac[par].append(self.param_init[par]*dsdp_extract[d][p]*self.scale_constant_value)
+                    else:
+                        jac[par].append(dsdp_extract[d][p]*self.scale_constant_value)
+
+            time11 = time.time()
+            if self.verbose:
+                print('Build time with direct kaug mode [s]:', time_build)
+                print('Solve time with direct kaug mode [s]:', time_solve)
+                print('Total wall clock time [s]:', time11-time00)
+                
+            # check if another prior experiment FIM is provided other than the user-specified one
+            if specified_prior is None:
+                prior_in_use = self.prior_FIM
+            else:
+                prior_in_use = specified_prior
+
+            # Assemble and analyze results
+            FIM_analysis = FIM_result(self.param_name,self.measure, jacobian_info=None, all_jacobian_info=jac,
+                                      prior_FIM=prior_in_use, store_FIM=FIM_store_name,
+                                      scale_constant_value=self.scale_constant_value)
+            
+            
+            self.jac = jac
+            FIM_analysis.build_time = time_build
+            FIM_analysis.solve_time = time_solve
+            
+
+            return FIM_analysis
+
+
+        else:
+            raise ValueError('This is not a valid mode. Choose from "sequential_finite", "simultaneous_finite", "sequential_sipopt", "sequential_kaug"')
```

Comment:
These mode names don't match the ones listed in the docstring. This again supports my previous comment about defining this list in one place and then referencing it everywhere else instead of manually typing these modes everywhere.